src/marshmallow/fields.py

```diff
@@ -1625,6 +1629,56 @@ def __init__(self, *args, **kwargs):
         self.validators.insert(0, validator)
 
 
+class IP(Field):
+    """A IP address field.
+
+    :param bool exploded: If `True`, serialize ipv6 address in long form, ie. with groups
+        consisting entirely of zeros included."""
+
+    default_error_messages = {"invalid_ip": "Not a valid IP address."}
+
+    DESERIALIZATION_CLASS = None  # type: typing.Optional[typing.Type]
```

Comment:
I'd write

    DESERIALIZATION_CLASS = ipaddress.ip_address

Then in `_deserialize`

            return self.DESERIALIZATION_CLASS(utils.ensure_text_type(value))

Unless there is a reason not to do that?

Reply:
I've presented rationale for this in the comment to the pull request. Long story short, 'ip_address' is a function and eagers to become a method when assigned to a class attribute.

Reply:
Sorry, I did not understand your PR comment.

As you say, it works if we use `staticmethod`:

```py 
DESERIALIZATION_FUNC = staticmethod(ipaddress.ip_address)
```

mypy doesn't complain if I write


```py 
    DESERIALIZATION_FUNC = staticmethod(
        ipaddress.ip_address
    )  # type: typing.Union[staticmethod, ipaddress.IPv4Address, ipaddress.IPv6Address]
```

but I have little experience with Typing so I can't tell if it is the right way to go.

Type checking aside, I found the staticmethod alternative better but I'm starting to understand your rationale.

(Slight performance advantage for the staticmethod alternative because the check is not performed every load, but that's peanuts.)